components/esmterrain/storage.cpp

```diff
@@ -448,14 +456,26 @@ namespace ESMTerrain
                     int layerIndex = textureIndicesMap.find(id)->second;
                     int blendIndex = (pack ? static_cast<int>(std::floor((layerIndex - 1) / 4.f)) : layerIndex - 1);
                     int channel = pack ? std::max(0, (layerIndex-1) % 4) : 0;
-
-                    if (blendIndex == i)
-                        pData[(blendmapSize - y - 1)*blendmapSize*channels + x*channels + channel] = 255;
+                    
+                    int alpha = (blendIndex == i) ? 255 : 0;
+                    
+                    if(!largeImage)
+                        pData[((blendmapSize - y - 1)*blendmapSize + x)*channels + channel] = alpha;
                     else
-                        pData[(blendmapSize - y - 1)*blendmapSize*channels + x*channels + channel] = 0;
+                    {
+                        int realY = (blendmapSize - y - 1)*imageScaleFactor;
+                        int realX = x*imageScaleFactor;
+                        if(true)
```

Comment:
Neither does this condition have any worth. I presume they were for debug.

Reply:
This one's for text alignment. Is that bad?

Reply:
Usually we do not use redundant conditions.
And I have a suspicion that this block really can be a some kind of "if-then-else".

Reply:
Now that I'm fully awake again I realized that the conditions aren't even necessary as long as blendmapImageSize is a multiple of blendmapSize. I wrote them when I was still trying to make blendmapImageSize = blendmapSize*2-1 work.